Tests/RevenueCatUITests/PaywallsV2/TakeScreenshot.swift

```diff
@@ -0,0 +1,172 @@
+import SnapshotTesting
+import SwiftUI
+
+#if !os(watchOS) && !os(macOS)
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+class TakeScreenshotTests: BaseSnapshotTest {
+
+    struct PackageData: Decodable {
+        let packages: [OfferingsResponse.Offering.Package]
+    }
+
+    func testPaywallValidationScreenshots() {
+        let bundle = Bundle(for: Self.self)
+
+        guard let resourceBundleURL = bundle.url(
+            forResource: "RevenueCat_RevenueCatUITests", withExtension: "bundle"
+        ) else {
+            fatalError("Could not locate RevenueCat_RevenueCatUITests.bundle")
+        }
+
+        let packagesPath = resourceBundleURL
+            .appendingPathComponent("__PreviewResources__")
+            .appendingPathComponent("resources")
+            .appendingPathComponent("packages.json")
+
+        let offeringsPath = resourceBundleURL
+            .appendingPathComponent("__PreviewResources__")
+            .appendingPathComponent("resources")
+            .appendingPathComponent("paywall-templates")
+            .appendingPathComponent("offerings_paywalls_v2_templates.json")
+
+        // Read and decode or print contents
+        guard let packagesData = try? Data(contentsOf: packagesPath) else {
+            fatalError("Couldn't parse packages data")
+        }
+        guard let packages = try? JSONDecoder.default.decode(PackageData.self, from: packagesData) else {
+            fatalError("Failed to decode packages data")
+        }
+        guard let offeringsData = try? Data(contentsOf: offeringsPath) else {
+            fatalError("Couldn't parse offerings data")
+        }
+        guard let offeringsResponse = try? JSONDecoder.default.decode(
+            OfferingsResponse.self, from: offeringsData
+        ) else {
+            fatalError("Failed to decode offerings data")
+        }
+
+        let offeringsWithPackages = offeringsResponse.offerings.map { offering in
+            return OfferingsResponse.Offering(
+                identifier: offering.identifier,
+                description: offering.description,
+                packages: packages.packages,
+                paywallComponents: offering.paywallComponents,
+                draftPaywallComponents: offering.draftPaywallComponents,
+                webCheckoutUrl: offering.webCheckoutUrl
+            )
+        }
+
+        let offeringsResponseWithPackages = OfferingsResponse(
+            currentOfferingId: offeringsResponse.currentOfferingId,
+            offerings: offeringsWithPackages,
+            placements: offeringsResponse.placements,
+            targeting: offeringsResponse.targeting,
+            uiConfig: offeringsResponse.uiConfig
+        )
+
+        let offerings = OfferingsFactory().createOfferings(from: [
+            "com.revenuecat.lifetime_product": .init(sk1Product: PreviewMock.Product(
+                price: 1.99,
+                unit: .week,
+                localizedTitle: "Liftime"
```

Comment:
I don't think it's important but just in case
```suggestion
                localizedTitle: "Lifetime"
```